### ISSUE TITLE
Disable refresh timer option

### ISSUE THREAD

**Issue by Willybits** [arkayenro/arkinventory#1056]
Hi!

Love your addon, been using it after engbags got useless :)

Would it be possible to disable the refreshtimer? At the moment when I am selling a lot of stuff, or putting this on the Auction House I missclick a lot because ark refreshes the bags and therefor the position of items change. I would like to only refresh when opening the bags, or requesting it manually ?

Thanks!

**Comment by arkayenro** [comment]
config > settings > design > pick the design being used for that location > window > sorting = manually (or on window open)

anything except instant

**Comment by Willybits** [comment]
Ahh, I was only looking at the update timers. Those were most in front of me. Thanks! I'll try it this evening :)

**Comment by Willybits** [comment]
I checked it out and all are on window open. Still the contents are refreshed when I am selling stuff after x seconds (I'm guessing the update timer)

**Comment by SamsaAsmas** [comment]
Every modification on the bag trigger an update (with open and manual refresh option).
Disenchant an item ? Get a new item ? The frame update
Sell something ? The frame update

This begins with the last update for me.

**Comment by arkayenro** [comment]
what exactly do you mean by "the frame update"?  the while bag recalculates and re-sorts everything? or something else?

**Comment by Willybits** [comment]
I think he means that the bag resorts and the spot where the item was that you sold is gone. Because of this the layout of the bags change during your actions. Thats how it is for me as well. It would be great if the icon would just be greyed out (or become empty) but remain there until you close the bag. This way, the bag won't move around while handling stuff in the bag

**Comment by arkayenro** [comment]
thats exactly how it works when you set it on manual sorting, the code no longer does any sorting at all until you click on the refresh button

all i can think of is that they are changing the setting in the wrong design, ie one that isnt assigned to that location

**Comment by Willybits** [comment]
I'l double check again, but it is not working for me as well, it always sorts.
I am checking here: 
In controls Backpack uses [1001] Default- Bag
In Settings [1001] Default- Bag is set at Sorting On Window open
To be sure I set every soring on window open and it still always sorts after selling an item

**Comment by Willybits** [comment]
Hey, I just noticed that there was no catagory set at the Controls > Backpack
After setting this at [9999] * Default it seems to work now

Could that be it? That after some update this is left at nothing ?

**Comment by arkayenro** [comment]
yes, there should always be something set for all three options.

it might go empty if you deleted the one that was in there, which means it should then go back to the default but its always best to have an active entry in there

**Comment by Willybits** [comment]
Allright, I'll just input default there. There is no entry at catagory for all views (Bag, Bank, Rep etc etc)

Thanks for the help!
Awesome bag addon, I cannot understand people just settling with onebag :)

**Comment by arkayenro** [comment]
i just checked, and yes, if you delete a design/categoryset that was assigned it will be blank when you next go look at the control options so it would revert to the system default values

**Comment by boddole** [comment]
I don't mean to fork this topic but I do think this is relevant:

I was having a related issue and after reading through this topic I just have to have this spelled out for me: What exactly is the purpose/function of General, Backpack, Custom (checked) and the time? Whether I have it unchecked, checked and 5 seconds, or checked for 0.1 seconds - I can't tell any difference in function. However, Settings, Window, Style, Sorting, When - makes perfect sense to me and the difference is obvious.

**Comment by arkayenro** [comment]
those are the "bucket" event/update timers that i've made configurable - there are others that are fixed and you cant change them.

for items/backpack typically im listening for the BAG_UPDATE event which is sent when you move / click / receive / lose an item, basically anything item related

i dont process the events instantly.  theyre queued up in those "buckets", and once the bucket is full (has reached its timer value) those events get processed, and the next lot are queue up.

basically every X.XX seconds i process each bucket, presuming theres something in there to process

the shorter the value for that location the more responsive it will feel, the larger the value the more laggy it will feel

the tradeoff is that the shorter it is the more processing that needs to be done to build the window.  if you have instant sorting enabled then it can chew up a lot of CPU if you move a lot of things around in a short amount of time

**Comment by arkayenro** [comment]
fixed in 3.08.08
